cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/response/stats/BrokerUtilizationStats.java

```diff
@@ -0,0 +1,137 @@
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.SortedMap;
+import java.util.concurrent.ConcurrentSkipListMap;
+
+
+/**
+ * Get broker level stats related to resource utilization in human readable format.
+ */
+public class BrokerUtilizationStats extends BrokerStats{
+  private static final String HOST = "Host";
+  private static final String HOSTS = "hosts";
+  private final List<SingleBrokerStats> _brokerStats;
+  private final SortedMap<String, BasicStats> _hostStats;
+
+  public BrokerUtilizationStats(KafkaCruiseControlConfig config) {
+    super(config);
+    _brokerStats = new ArrayList<>();
+    _hostStats = new ConcurrentSkipListMap<>();
+  }
+
+  public void addSingleBrokerUtilizationStats(String host, int id, Broker.State state, double diskUtil, double cpuUtil, double leaderBytesInRate,
+                                   double followerBytesInRate, double bytesOutRate, double potentialBytesOutRate,
+                                   int numReplicas, int numLeaders, boolean isEstimated, double capacity) {
+
+    SingleBrokerStats singleBrokerStats =
+        new SingleBrokerStats(host, id, state, diskUtil, cpuUtil, leaderBytesInRate, followerBytesInRate, bytesOutRate,
+                              potentialBytesOutRate, numReplicas, numLeaders, isEstimated, capacity);
+    _brokerStats.add(singleBrokerStats);
+    _hostFieldLength = Math.max(_hostFieldLength, host.length());
+    _hostStats.computeIfAbsent(host, h -> new BasicStats(0.0, 0.0, 0.0, 0.0,
+                                                         0.0, 0.0, 0, 0, 0.0))
+              .addBasicStats(singleBrokerStats.basicStats());
+    _isBrokerStatsEstimated = _isBrokerStatsEstimated || isEstimated;
+  }
+
+  @Override
+  protected void discardIrrelevantResponse() {
+    // Discard irrelevant response.
+    _brokerStats.clear();
+    _hostStats.clear();
+  }
+
+  /**
+   * Return an object that can be further be used to encode into JSON
+   */
+  @Override
+  public Map<String, Object> getJsonStructure() {
+    List<Map<String, Object>> hostStats = new ArrayList<>(_hostStats.size());
+
+    // host level statistics
+    for (Map.Entry<String, BasicStats> entry : _hostStats.entrySet()) {
+      Map<String, Object> hostEntry = entry.getValue().getJSONStructure();
+      hostEntry.put(HOST, entry.getKey());
+      hostStats.add(hostEntry);
+    }
+
+    // broker level statistics
+    List<Map<String, Object>> brokerStats = new ArrayList<>(_brokerStats.size());
+    for (SingleBrokerStats stats : _brokerStats) {
+      Map<String, Object> brokerEntry = stats.getJSONStructure();
+      brokerStats.add(brokerEntry);
+    }
+
+    // consolidated
+    Map<String, Object> stats = new HashMap<>(2);
+    stats.put(HOSTS, hostStats);
+    stats.put(BROKERS, brokerStats);
+    return stats;
+  }
+
+  @Override
+  public void discardIrrelevantResponse(CruiseControlParameters parameters) {
```

Comment:
consider using `return;` early on to prevent nested `if`'s

Reply:
We shouldn't change the behavior of the function - if we add `return;` we are not doing the nested `if` check (which is necessary because it ensures we are maintaining the cached json correctly).

Reply:
Oh oops, my bad! I didn't know that line 91 existed until I took a close look at it. I'll mark this as resolved.